sympy/solvers/bivariate.py

```diff
@@ -126,6 +138,8 @@ def _lambert(eq, x, domain=S.Complexes):
     where X = g(x) and x = g^-1(X), return the Lambert solution,
         ``x = g^-1(-c/b + (a/d)*W(d/(a*b)*exp(c*d/a/b)*exp(-f/a)))``.
     """
+    global lambertReal
+    lambertReal = None
```

Comment:
No global variables

Reply:
is there any other means to access variables across the functions so that to avoid global variables?

Reply:
The normal was is by using function parameters and arguments. I don't know what this variable is for but is it not just possible to check the `domain` argument?

Reply:
>  I don't know what this variable is for but is it not just possible to check the domain argument?
This case is for if there are symbols in the equation but the lambert function is in a real range...

i.e.

```
b = Symbol('b')
eq = 3*log(a**(3*x + 5)) + b*log(a**(3*x + 5)) + a**(3*x + 5)
assert solveset_real(eq, x) == Intersection(FiniteSet(Rational(-5, 3) - LambertW(Rational(1, 3))/(3*log(p))), Interval(0, oo))

```
Here, ` LambertW(Rational(1, 3))` is real.